ui/lib/core/addon/decorators/confirm-leave.js

```diff
@@ -20,34 +48,44 @@ export function withConfirmLeave() {
     return class ConfirmLeave extends SuperClass {
       @service store;
 
+      _rollbackModel(modelPath) {
+        const model = this.controller.get(modelPath);
+        if (!model || !model.hasDirtyAttributes || model.isSaving) {
+          // we only want to complete rollback if the model is dirty and not saving
+          return;
+        }
+        const method = model.isNew ? 'unloadRecord' : 'rollbackAttributes';
+        model[method]();
+        return;
```

Comment:
Would it be easier to reason to flip this around and perform the method in the condition rather than having the empty return?
```suggestion
        // we only want to complete rollback if the model is dirty and not saving
        if (model && model.hasDirtyAttributes && !model.isSaving) {
          const method = model.isNew ? 'unloadRecord' : 'rollbackAttributes';
          model[method]();
        }
```